src/systems/systems/src/navigation/radio_altimeter.rs

```diff
@@ -0,0 +1,864 @@
+use crate::failures::{Failure, FailureType};
+use crate::shared::arinc429::{Arinc429Word, SignStatus};
+use crate::shared::{random_from_range, ConsumePower, ElectricalBusType, ElectricalBuses};
+use crate::simulation::{
+    InitContext, Read, SimulationElement, SimulationElementVisitor, SimulatorReader,
+    SimulatorWriter, UpdateContext, VariableIdentifier, Write,
+};
+use std::time::Duration;
+use uom::si::angle::degree;
+use uom::si::f64::{Angle, Length, Power, Ratio, Time, Velocity};
+use uom::si::length::foot;
+use uom::si::power::watt;
+use uom::si::ratio::ratio;
+use uom::si::velocity::meter_per_second;
+
+/// This struct describes the installation of a physical Antenna in relation to an  aircraft's
+/// center of gravity.
+#[derive(Debug, Copy, Clone, PartialEq)]
+pub struct AntennaInstallation {
+    y: Length, // negative towards top (from CG), positive towards bottom (from CG)
+    z: Length, // negative towards front (from CG), positive for aft
+    electric_length: Length, // The electric length of the cables and antenna
+}
+
+impl AntennaInstallation {
+    pub fn new(y: Length, z: Length, electric_length: Length) -> Self {
+        Self {
+            y,
+            z,
+            electric_length,
+        }
+    }
+}
+
+/// This trait describes a transceiver pair as used in radio altimeters, usually in the C band.
+/// It models the physical transmission of the signal between the transceivers and returns the
+/// response.
+trait TransceiverPair {
+    fn response(&self) -> Option<TransceiverPairResponse>;
+}
+
+/// This struct encodes a response from a pair of transceivers. Currently it's only encodes as a
+/// travel time, which in an modulated continuous wave radar (FMCW) can already be considered the
+/// filtered duration between the transmission and the response. In future it may also return the
+/// remaining signal strength, which can be used to model the signal loss over various surfaces, or
+/// model an actual frequency response.
+#[derive(Debug, Copy, Clone, PartialEq)]
```

Comment:
**thought**: This thought only relates to our codebase being a "library" for others; which isn't the case yet (and in this case actually the type is not even `pub`, so it doesn't matter much). `Copy`, `Clone` and `PartialEq` are not used. While having `Clone` and `PartialEq` on a type is fine. One has to be careful with `Copy` as future changes might make the type no longer `Copy` (i.e. if it starts to contain any type which is not `Copy`). Removing it later is a breaking change. I think you'll only add things that will be `Copy` to this type, but in general this is something to be aware of.